drivers/led/ncp5623c.c

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2023 NXP Semiconductors
+ * Copyright (c) 2023 Cognipilot Foundation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT onnn_ncp5623c
+
+/**
+ * @file
+ * @brief NCP5623C LED driver
+ *
+ * The NCP5623C is a 3-channel LED driver that communicates over I2C.
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(ncp5623c, CONFIG_LED_LOG_LEVEL);
+
+#include "led_context.h"
+
+#define NCP5623_LED_CURRENT 0x20
+#define NCP5623_LED_PWM0    0x40
+#define NCP5623_LED_PWM1    0x60
+#define NCP5623_LED_PWM2    0x80
+
+#define NCP5623C_CHANNEL_COUNT 3
+
+/* Brightness limits in percent */
+#define NCP5623C_MIN_BRIGHTNESS 0
+#define NCP5623C_MAX_BRIGHTNESS 0x1f
+
+static const uint8_t led_channels[] = {NCP5623_LED_PWM0, NCP5623_LED_PWM1, NCP5623_LED_PWM2};
+
+struct ncp5623c_config {
+	struct i2c_dt_spec bus;
+	uint8_t num_leds;
+	const struct led_info *leds_info;
+};
+
+struct ncp5623c_data {
+	struct led_data dev_data;
+};
+
+static const struct led_info *ncp5623c_led_to_info(const struct ncp5623c_config *config,
+						   uint32_t led)
+{
+	int i;
+
+	for (i = 0; i < config->num_leds; i++) {
+		if (config->leds_info[i].index == led) {
+			return &config->leds_info[i];
+		}
+	}
+
+	return NULL;
```

Comment:
Please replace the code above with:

```
if (led < config->num_leds) {
	return &config->leds_info[led];
}

return NULL;
```

See this [patch](https://github.com/zephyrproject-rtos/zephyr/pull/59852/commits/d36b6dbc7f5ba50ce1d2448c84a4309ae84d320a) for the explanation.